lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,28 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "the value of 'other' MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
+
+      IncludeDirective.new(hash, options.merge!(other.options))
```

Comment:
The problem here is that if one has two `IncludeDirectives` with conflicting options, it is not clear to the user how this conflict will be resolved. In this case it's the options of the `other` hash that will have priority.